File: src/main/kotlin/tryout/moduleconfig/pushToFunction/bar-bf.ts
/*
 * Copyright © 2022 Paulo Villela. All rights reserved.
 * Use of this source code is governed by the MIT license
 * that can be found in the LICENSE file.
 */

export type BarBfCfgInfo = {
  z: number;
};

export type BarBfCfgSrc = {
  get: () => BarBfCfgInfo;
};

export type BarBfT = () => void;

export function barBfC(cfg: BarBfCfgSrc): BarBfT {
  return function() {
    console.log(cfg.get().z);
  }
}

File: src/main/kotlin/tryout/moduleconfig/pushtovar/fs/barBf.kt
/*
 *  Copyright © 2021 Paulo Villela. All rights reserved.
 *  Use of this source code is governed by the Apache 2.0 license
 *  that can be found in the LICENSE file.
 */

package tryout.moduleconfig.pushtovar.fs

import tryout.moduleconfig.pullwithpushoverride.fwk.CfgSrc
import tryout.moduleconfig.pullwithpushoverride.fwk.makeConfigSource

data class BarBfCfgInfo(
	val z: Int
)

val barBfCfgSrc: CfgSrc<BarBfCfgInfo> = makeConfigSource()

fun barBf() {
	println(barBfCfgSrc.get().z)
}
